configure.ac

```diff
@@ -1,313 +1,127 @@
-    [:],
-    [enable_flint=check])
-AS_IF([test "x$enable_flint" != xno],
-    [FLINT_LIBS="-lflint -lmpfr"
-     LIBS_SAVED="$LIBS"
-     LIBS="$LIBS $FLINT_LIBS"
-     AC_CHECK_LIB(flint,dummy_fn_to_prevent_ISOC_warning,[have_flint=yes],[have_flint=no])
-     LIBS="$LIBS_SAVED"
-    ])
-AS_IF([test x$have_flint = xyes],
-  [enable_flint=yes],
-  [AS_IF([test "x$enable_flint" = xyes],
-    [AC_MSG_ERROR([flint is not available])],
-    [AS_IF([test "x$enable_flint" != xno],
-        [AC_MSG_NOTICE([flint will not be used because flint could not be found])])
-     enable_flint=no])])
-AC_SUBST(FLINT_LIBS)
-AC_SUBST(COCOA_CPPFLAGS)
-AM_CONDITIONAL(NMZ_FLINT, test "x$enable_flint" = xyes)
-AS_IF([test "x$enable_flint" = xyes],
-  [AC_DEFINE(NMZ_FLINT)
-   DEFINE_NMZFLINT="#define NMZ_FLINT"
-   AC_SUBST([DEFINE_NMZFLINT])])
+# We can optionally build with FLINT
+AC_ARG_WITH([flint], AS_HELP_STRING([--with-flint], [Use FLINT]))
+AS_IF([test "x$with_flint" = "x"], [with_flint=no])
+AS_IF([test "x$with_flint" != "xno"], [
+    AC_CHECK_LIB([flint], [fmpz_init], [], [AC_MSG_ERROR([libflint not found.])], [-lmpfr])
+    LIBS="-lmpfr $LIBS"
+    AC_DEFINE(NMZ_FLINT)
+    DEFINE_NMZFLINT="#define NMZ_FLINT"
+    AC_SUBST([DEFINE_NMZFLINT])
+])
+AM_CONDITIONAL(NMZ_FLINT, test "x$with_flint" != "xno")
   
-### Test whether to build Normaliz with nauty.
-AC_ARG_WITH([nautylib],
-   AS_HELP_STRING([--with-nauty=DIR],
-      [provide location of nauty installation prefix or source directory]),
-   [ LDFLAGS="-L$with_nauty/lib $LDFLAGS"
-     CPPFLAGS="-I $with_nauty/include $CPPFLAGS"
-   ])
-AC_ARG_ENABLE([nmzinzautomorph],
-    [AS_HELP_STRING([--enable-nmzautomorph@<:@=ARG@:>@],
-      [enable nmzInautomorph (requires nauty) @<:@default=check@:>@])],
-    [:],
-    [enable_nmznauty=check])
-AS_IF([test "x$enable_nmznauty" != xno],
-    [AC_MSG_CHECKING([whether nauty headers and library are available])
-     NAUTY_LIBS="-lnauty"
-     LIBS_SAVED="$LIBS"
-     LIBS="$LIBS $NAUTY_LIBS"
-     AC_CHECK_LIB(nauty,densenauty,[have_nauty=yes],[have_nauty=no])
-     AC_MSG_RESULT($have_nauty)
-     LIBS="$LIBS_SAVED"
-    ])
-AS_IF([test x$have_nauty = xyes],
-  [enable_nmznauty=yes],
-  [AS_IF([test "x$enable_nmznauty" = xyes],
-    [AC_MSG_ERROR([nauty is not available and is required for automorphisms])],
-    [AS_IF([test "x$enable_nmzintegrate" != xno],
-        [AC_MSG_NOTICE([normaliz builzauztomorphisms  built because nauty was not found])])
-     enable_nmznauty=no])])
-AC_SUBST(NAUTY_LIBS)
-AC_SUBST(NAITY_CPPFLAGS)
-AM_CONDITIONAL(NMZ_NAUTY, test x$enable_nmznauty = xyes)
-AS_IF([test "x$have_nauty" = xyes],
-  [AC_DEFINE(NMZ_NAUTY)
-   DEFINE_NMZNAUTY="#define NMZ_NAUTY"
-   AC_SUBST([DEFINE_NMZNAUTY])])
+# We can optionally build with nauty.
+AC_ARG_WITH([nauty], AS_HELP_STRING([--with-nauty], [Use nauty for nmzInautomorph]))
+AS_IF([test "x$with_nauty" = "x"], [with_nauty=no])
+AS_IF([test "x$with_nauty" != "xno"], [
```

Comment:
But you never use the value of `with_nauty`. If you don't want to support specifying custom installation paths for dependencies anymore (but why?), then shouldn't this just use `AC_ARG_ENABLE` instead? (Same of course applies to virtually all the other dependencies).

Reply:
The (probably unwritten) standard says that `--enable` is for enabling/disabling features that are usually independent of dependencies. `--with` is for building against dependencies. The value of `--with` is usually not used. You just look for `with/without` and ignore the actual value typically.

Reply:
You are right, the GNU autoconf manual explicitly writes (so it is written :-)) that `--with` is for packages/dependencies, and `--enable` for feature. OK, fine.

But that the value of `--with` is usually not used is simply not true, based on dealing with this stuff for the past twenty years. For example, let's look at the GCC configure -- arguably, GCC is maybe *the* GNU project. It does this:
```
...

Optional Packages:
  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
  --with-build-libsubdir=DIR  Directory where to find libraries for build system
  --with-system-zlib      use installed libz
  --with-mpc=PATH         specify prefix directory for installed MPC package.
                          Equivalent to --with-mpc-include=PATH/include plus
                          --with-mpc-lib=PATH/lib
  --with-mpc-include=PATH specify directory for installed MPC include files
  --with-mpc-lib=PATH     specify directory for the installed MPC library
  --with-mpfr-dir=PATH    this option has been REMOVED
  --with-mpfr=PATH        specify prefix directory for installed MPFR package.
                          Equivalent to --with-mpfr-include=PATH/include plus
                          --with-mpfr-lib=PATH/lib
  --with-mpfr-include=PATH
                          specify directory for installed MPFR include files
  --with-mpfr-lib=PATH    specify directory for the installed MPFR library
  --with-gmp-dir=PATH     this option has been REMOVED
  --with-gmp=PATH         specify prefix directory for the installed GMP
                          package. Equivalent to
                          --with-gmp-include=PATH/include plus
                          --with-gmp-lib=PATH/lib
  --with-gmp-include=PATH specify directory for installed GMP include files
  --with-gmp-lib=PATH     specify directory for the installed GMP library
  --with-stage1-libs=LIBS libraries for stage1
  --with-stage1-ldflags=FLAGS
                          linker flags for stage1
  --with-boot-libs=LIBS   libraries for stage2 and later
  --with-boot-ldflags=FLAGS
                          linker flags for stage2 and later
  --with-isl=PATH         Specify prefix directory for the installed isl
                          package. Equivalent to
                          --with-isl-include=PATH/include plus
                          --with-isl-lib=PATH/lib
  --with-isl-include=PATH Specify directory for installed isl include files
  --with-isl-lib=PATH     Specify the directory for the installed isl library
...
```

Of course one can still say that one does not want to support a prefix argument. And if this `configure.ac` file was being created new and from scratch, I'd be fine with that. But you are not proposing to remove functionality, without a clear argument for why, i.e., what is gained by this. At least it didn't become clear to me, but perhaps you can elaborate?

Reply:
GCC uses these paths because GCC is very complex (different parts of GCC build against varying dependencies I guess.) Most projects do not use this afaik.

The problem is that `--with-gmp=path` is an alias for `LDFLAGS=-Lpath/lib` `CPPFLAGS=`-Ipath/include`. I think that people should just set their LDFLAGS  and CPPFLAGS correctly; that's certainly how distributions build packages. We can keep the current behaviour but it creates a fair bit of complexity in the configure.ac (you need to handle no and yes differently and patch LDFLAGS and CPPFLAGS.) Since we're breaking the interface anyway by removing the `--enable` flags, we can also change this I believe.

Note that `--with-flint=$PREFIX` (that's what existing build scripts for distributions will do) still works. The argument is ignored but their LDFLAGS and CPPFLAGS certainly include the correct paths already anyway.